src/blocks/libvirt.rs

```diff
@@ -73,51 +73,95 @@ impl ConfigBlock for Libvirt {
 
 impl Block for Libvirt {
     fn update(&mut self) -> Result<Option<Update>> {
-        if !self
+        // Check if the connection is still active and re-instatiate it if it's not
+        match self
             .qemu_conn
             .is_alive()
-            .expect("unrecoverable error with qemu object")
+            .block_error("virt", "error when getting connection status to libvirt")
         {
-            self.qemu_conn = Connect::open_read_only(
-                &self
-                    .qemu_conn
-                    .get_uri()
-                    .expect("could not get URI from the currently connected QEMU object"),
-            )
-            .expect("could not re-connect to libvirtd")
+            Ok(true) => {}
+            Ok(false) => {
+                self.qemu_conn = match Connect::open_read_only(
+                    &self
+                        .qemu_conn
+                        .get_uri()
+                        .block_error("virt", "error in retrieving URI information from libvirt")
+                        .unwrap(), /* Only happens if memory corruption; then we should panic anyway */
+                )
+                .block_error("virt", "error in reconnecting to libvirt socket")
+                {
+                    Ok(c) => c,
+                    Err(e) => return Err(e),
+                };
```

Comment:
Rust has a syntactic shortcut for this.

```rust
let a = match b {
    Ok(c) => c,
    Err(e) => return Err(e),
};
```

is the same as

```rust
let a = b?;
```

Reply:
I intentionally avoided the `?` operator as the Error type (`virt::Error::error`) does not go into Error::error properly, but after checking it seems to be working, so I'm not sure how I ended up coming to that conclusion.

Thanks for the suggestion. :) <3